crates/bevy_tasks/src/pollable_task.rs

```diff
@@ -0,0 +1,33 @@
+use crate::Task;
+use async_channel::{Receiver, TryRecvError};
+
+/// A pollable task whose result readiness can be checked in system functions
+/// on every frame update without blocking on a future
+#[derive(Debug)]
+pub struct PollableTask<T> {
+    receiver: Receiver<T>,
+    // this is to keep the task alive
+    _task: Task<()>,
+}
+
+impl<T> PollableTask<T> {
+    pub(crate) fn new(receiver: Receiver<T>, task: Task<()>) -> Self {
+        Self {
+            receiver,
+            _task: task,
+        }
+    }
+
+    /// poll to see whether the task finished
+    pub fn poll(&self) -> Option<T> {
+        match self.receiver.try_recv() {
+            Ok(value) => Some(value),
+            Err(try_error) => match try_error {
+                TryRecvError::Empty => None,
+                TryRecvError::Closed => {
+                    panic!("Polling on the task failed because the connection was already closed.")
```

Comment:
Is this panic-worthy? I'm not sure exactly when the actual spawned future is dropped, but I think it would be reasonable for the executor to drop it once it reaches `Ready` once.

I'd say it might be better to add e.g. an `AtomicBool` to this object, and print a warning in this case, at most once for each task. I'm not sure of the exact details of the executor though

Reply:
Hmm, it seems quite harsh to stop the world for running the check after the task has ended.

I don't want to block on this; I don't feel very strongly either way.